homeassistant/components/asuswrt/config_flow.py

```diff
@@ -166,17 +183,28 @@ async def async_step_user(
                 errors["base"] = "invalid_host"
 
         if not errors:
-            result = await self._async_check_connection(user_input)
-            if result != RESULT_SUCCESS:
-                errors["base"] = result
-
-        if errors:
-            return self._show_setup_form(user_input, errors)
-
-        return self.async_create_entry(
-            title=host,
-            data=user_input,
-        )
+            result, unique_id = await self._async_check_connection(user_input)
+            if result == RESULT_SUCCESS:
+                if unique_id:
+                    await self.async_set_unique_id(unique_id)
+                    self._abort_if_unique_id_configured()
```

Comment:
We should only abort if this is from discovery. Since this is user configured, we should continue. They might want to update data. All existing data is pointed at new one. 
```suggestion
```

Reply:
Done, and updated tests accordingly.